tests/features/workspace.feature

```diff
@@ -0,0 +1,153 @@
+    And   I call workspace components method
+    Then  I expect to receive a map of promises to component 1 and 2 with the following <property>s
+          |<property>   |
+          |componentName|
+          |path         |
+          |config       |
+
+Scenario: Call registerService method registers the provided service in the Workspace
+    Given WorkspaceFactory instance with createWorkspace method
+    And  Configuration for token 123 that includes service A and B and components 1 and 2
+    And  Service A and service B include a bootstrap that call registerService
+    When I run createWorkspace method with token 123 and Workspace is created
+    And  I call registerService method with service A and with a valid request 
+	And  The request has the following <property> and <type>
+         |<property> | <type>|
+         |serviceName| string|
+         |displayName| string|
+         |reference  | any   |
+    Then Service A is registered in the Workspace
+
+Scenario: Call registerService method with a service already registered is rejected with error
+    Given WorkspaceFactory instance with createWorkspace method
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B include a bootstrap that calls registerService
+    And   I run createWorkspace with token 123 and Workspace is created
+    And   I call registerService method with service A and service is registered
+	When  I call registerService method again with service A
+    Then  I expect to receive an error
+
+Scenario: Call registerService method with an invalid serviceName is rejected with error
+    Given WorkspaceFactory instance with createWorkspace method
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I run createWorkspace with token 123 and Workspace is created
+    And   I call workspace registerService method with invalid values for <serviceName> and valid displayName
+          |<serviceName> |
+          |''        |
+          |{}        |
+          |{ test: 'test' }|
+          |[]        |
+          |['test']  |
+          |null      |
+          |undefined |
+          |true      |
+          |false     |
+          |0         |
+          |-1        |
+    Then  I expect to receive an error
+
+Scenario: Call registerService method with an invalid displayName is rejected with error
+    Given WorkspaceFactory instance with createWorkspace method
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I run createWorkspace with token 123 and Workspace is created
+    And   I call workspace registerService method with invalid values for <displayName> and valid serviceName
```

Comment:
We should remove all occurrences of `displayName`, it has been removed from the model